azplugins/flow.py

```diff
@@ -717,25 +722,41 @@ def __call__(self, timestep):
 
         if hoomd.comm.get_rank() == 0:
             x = snap.particles.position[:,self.bin_axis]
-            v = snap.particles.velocity[:,self.flow_axis]
+            v = snap.particles.velocity
+            m = snap.particles.mass
 
-            _counts,_ = np.histogram(x, bins=self.bins, range=self.range)
-            self._counts += _counts
+            binids = np.digitize(x, self.edges[1:])
```

Comment:
This seems a little funny because it is discarding the particles in the first bin (and making them fall in some error case). As I read the documentation, I think we want to use all the edges with `right=False` (the default), check if any of the bins are 0 (which would be an error), then subtract 1 from the bin so that they are 0-indexed. Do you agree?

Reply:
Questions to consider:

- should we check that the range is inside of the box? 
- if the user specifies a range smaller than the box, particles could be outside of the range, so maybe that case should not be an error, but those particles should be filtered out/ignored?

Reply:
for now: no checks if `range` is inside box, added comment to documentation.